metaverse_components/pet.js

```diff
@@ -90,45 +141,106 @@ export default (app, component) => {
       _unwear();
     }
   });
-  
+
   const smoothVelocity = new THREE.Vector3();
   const lastLookQuaternion = new THREE.Quaternion();
   const _getAppDistance = () => {
-    const position = localVector.copy(player.position);
-    position.y = 0;
+    const position = player.position.clone();
+
+    // setting the height of the y as the height of the ground
+    const collisionToGround = physics.raycast(position, downQuaternion);

```

Comment:
This seems like it can be entirely resolved by having correct character controller capsule and offset settings without any need for an external raycast.

Reply:
This code is for finding the ground y position ( it's used for follow functionality )